src/Microsoft.Azure.ServiceBus/Message.cs

```diff
@@ -192,22 +196,17 @@ public DateTime ExpiresAtUtc
         ///      setting and it is silently adjusted if it does. 
         ///      See <a href="https://docs.microsoft.com/azure/service-bus-messaging/message-expiration">Expiration</a>
         /// </remarks>
-        public TimeSpan TimeToLive
+        public TimeSpan? TimeToLive
         {
-            get
-            {
-                if (this.timeToLive == TimeSpan.Zero)
-                {
-                    return TimeSpan.MaxValue;
-                }
-
-                return this.timeToLive;
-            }
+            get => this.timeToLive;
 
             set
             {
-                TimeoutHelper.ThrowIfNonPositiveArgument(value);
-                this.timeToLive = value;
+                if (value != null)
+                {
+                    TimeoutHelper.ThrowIfNonPositiveArgument(value.Value);
```

Comment:
Should there be another helper method on `TimeoutHelper` to take a nullable TimeSpan? It looks odd to test for `null` and yet call a guard method after.